### PR TITLE
review additional changes before commiting to master

### PR DESCRIPTION
**Pull request by Darkryhr**

### CODE REVIEW

File: client/src/app/services/order.service.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export class OrderService {
 
   add(product: string, amount: number, price: number) {
```
Comment (alexrubin73) [inline client/src/app/services/order.service.ts]: return type

File: client/src/app/services/order.service.ts [hunk 2]
```diff
@@ -48,32 +48,31 @@ export class OrderService {
     this.cartSubject.next([]);
   }
 
+  //* Get full cart
   getAllItems() {
```
Comment (alexrubin73) [inline client/src/app/services/order.service.ts:63]: return type

File: client/src/app/services/order.service.ts [hunk 3]
```diff
@@ -48,32 +48,31 @@ export class OrderService {
...
     return this.httpClient.get<OrderResponse>(this.baseUrlOrder);
   }
 
+  getOrder(id) {
```
Comment (alexrubin73) [inline client/src/app/services/order.service.ts]: return type

File: client/src/app/services/order.service.ts [hunk 4]
```diff
@@ -48,32 +48,31 @@ export class OrderService {
...
+    return this.httpClient.get(`${this.baseUrlOrder}/invoice/${id}`);
+  }
+
   getTotalCarts() {
```
Comment (alexrubin73) [inline client/src/app/services/order.service.ts:73]: return type

File: client/src/app/services/order.service.ts [hunk 5]
```diff
@@ -48,32 +48,31 @@ export class OrderService {
...
-    return this.httpClient.get(`${this.baseUrl}/sum`);
+    return this.httpClient.get(`${this.endpoint}/sum`);
   }
 
   completeOrder(body) {
```
Comment (alexrubin73) [inline client/src/app/services/order.service.ts:77]: return type

File: client/src/app/services/order.service.ts [hunk 6]
```diff
@@ -48,32 +48,31 @@ export class OrderService {
...
-          this.clear();
-          this.openInvoiceDialog(res.data.order._id);
-          this.router.navigate(['products']);
-        } else {
-          // send error
-        }
+        //TODO: handle error
+        if (res.message !== 'success') return;
+        this.clear();
+        this.openInvoiceDialog(res.data.order._id);
```
Comment (alexrubin73) [inline client/src/app/services/order.service.ts:84]: is it ok to open a dialog and navigate at the same time?

File: client/src/app/services/order.service.ts (hunk 2 above)
Comment (Darkryhr) [reply client/src/app/services/order.service.ts:63]: can't I use type inference here?

File: client/src/app/services/order.service.ts (hunk 6 above)
Comment (Darkryhr) [reply client/src/app/services/order.service.ts:84]: the dialog is a modal, if you click out, it closes

File: client/src/app/services/order.service.ts (hunk 5 above)
Comment (Darkryhr) [reply client/src/app/services/order.service.ts:77]: added generic

File: client/src/app/services/order.service.ts (hunk 2 above)
Comment (alexrubin73) [reply client/src/app/services/order.service.ts:63]: OrderResponse ?

File: client/src/app/services/order.service.ts (hunk 5 above)
Comment (alexrubin73) [reply client/src/app/services/order.service.ts:77]: i dont see it... did you push the commit?

File: client/src/app/services/order.service.ts (hunk 6 above)
Comment (alexrubin73) [reply client/src/app/services/order.service.ts:84]: and the routed page is in the background ?

Comment (Darkryhr) [reply client/src/app/services/order.service.ts:84]: yep

File: client/src/app/services/order.service.ts (hunk 5 above)
Comment (Darkryhr) [reply client/src/app/services/order.service.ts:77]: yes, maybe it doesnt show it on here

File: client/src/app/services/order.service.ts (hunk 2 above)
Comment (Darkryhr) [reply client/src/app/services/order.service.ts:63]: yes - Observable<OrderResponse>

**Pull request merged**